crates/data_components/src/arrow/write.rs

```diff
@@ -17,19 +17,22 @@
 
 //! [`MemTable`] for querying `Vec<RecordBatch>` by `DataFusion`.
 
+use arrow::array::BooleanBuilder;
+use arrow::compute::filter_record_batch;
 use datafusion::catalog::Session;
 // This is modified from the DataFusion `MemTable` to support overwrites. This file can be removed once that change is upstreamed.
```

Comment:
```suggestion
```

I'm guessing we're probably not going to upstream it at this point.

Reply:
What's stopping us (besides getting interest about our changes?)

Reply:
If we don't remove the comment then we can modify to include the changes we've made.